backend/pkg/helm/repository.go

```diff
@@ -0,0 +1,261 @@
+package helm
+
+import (
+	"encoding/json"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	zlog "github.com/rs/zerolog/log"
+	"helm.sh/helm/v3/pkg/getter"
+	"helm.sh/helm/v3/pkg/repo"
+)
+
+const (
+	defaultNewConfigFileMode   os.FileMode = os.FileMode(0644)
+	defaultNewConfigFolderMode os.FileMode = os.FileMode(0770)
+)
+
+// add repository
+
+type AddUpdateRepoRequest struct {
+	Name string `json:"name"`
+	URL  string `json:"url"`
+	// TODO: Figure out how to support auth
+	// like username, password, certfile etc
+	// https://github.com/helm/helm/blob/39ca699ca790e02ba36753dec6ba4177cc68d417/cmd/helm/repo_add.go#L169
+}
+
+// Creates a filename if it's not there, including any missing directories.
+func createFileIfNotThere(fileName string) error {
+	_, err := os.Stat(fileName)
+	if os.IsNotExist(err) {
+		// create changes
+		_, err = createFullPath(fileName)
+		return err
+	}
+
+	return nil
+}
+
+// Adds a repository with name, url to the helm config. Returns error if there is one.
+func AddRepositoryToHelm(name string, url string) error {
+	err := createFileIfNotThere(settings.RepositoryConfig)
+	if err != nil {
+		zlog.Error().Err(err).Str("action", "add_repo").Msg("failed to create empty RepositoryConfig file")
+	}
+
+	// TODO: Lock repo file
```

Comment:
There's a TODO here.

Reply:
This is fine to improve later IMO.

Reply:
It's not clear to me if this could cause corruption of helm config if more than one helm code caller runs at the same time (either multiple headlamp actions or helm binary whilst headlamp is doing actions). The github linked shows locking, and it seems to me this is required by API users. Needs further investigation.

Reply:
Since the file lock is added there won't be corruption of helm repo file due to concurrency. Are there any other issues that are on your mind that needs to be investigated?

Reply:
I noticed the helm binary doesn’t lock on anything but the add repo call. Which is a bug in helm. Probably we should lock it everywhere it modifies. Probably locking is more important with a web app because it’s easier to fire off multiple requests (double clicks on buttons, impatient users etc).

Reply:
AFAIU Apart from the add repo and update repo most of the functions are read only. Ill check once if there are any other places where we need to add locks.

Reply:
It looks like @yolossn added locks to the other other methods. 🎉  So resolving this convo.